nonce/nonce.go

```diff
@@ -27,9 +27,7 @@ import (
 	"github.com/letsencrypt/boulder/metrics"
 )
 
-// MaxUsed defines the maximum number of Nonces we're willing to hold in
-// memory.
-const MaxUsed = 65536
+const defaultMaxUsed = 65536
```

Comment:
tiny nit: you could combine the `defaultMaxUsed` and `nonceLen` `const`s into one block:
```
const (
  defaultMaxUsed = 65536
  nonceLen = 32
)
```